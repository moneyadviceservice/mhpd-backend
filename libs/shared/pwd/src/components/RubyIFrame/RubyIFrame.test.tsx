import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { act } from 'react-dom/test-utils';
import { RubyIFrame } from '.';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

describe('RubyIFrame component', () => {
  it('renders correctly', () => {
    render(
      <RubyIFrame
        toolData={{
          url: {
            en: 'https://embedded-journeys.moneyhelper.org.uk/en/leave-pot-untouched',
            cy: 'https://embedded-journeys.moneyhelper.org.uk/cy/leave-pot-untouched',
          },
          id: 'leave-pot-untouched',
        }}
        testId="test-component"
      />,
    );
    const container = screen.getByTestId('test-component');
    expect(container).toMatchSnapshot();
  });

  it('updates iframe height on MASRESIZE- message', async () => {
    render(
      <RubyIFrame
        toolData={{
          url: {
            en: 'https://embedded-journeys.moneyhelper.org.uk/en/leave-pot-untouched',
            cy: 'https://embedded-journeys.moneyhelper.org.uk/cy/leave-pot-untouched',
          },
          id: 'leave-pot-untouched',
        }}
      />,
    );
    const container = screen.getByTestId('iframe');
    act(() => {
      window.dispatchEvent(
        new MessageEvent('message', {
          data: 'MASRESIZE-200',
        }),
      );
    });

    await waitFor(() => {
      expect(container).toHaveAttribute('height', '200');
    });
  });
});
