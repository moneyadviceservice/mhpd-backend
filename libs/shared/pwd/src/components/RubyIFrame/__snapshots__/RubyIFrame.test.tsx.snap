// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RubyIFrame component renders correctly 1`] = `
<iframe
  class="w-full"
  data-testid="test-component"
  height="0"
  id="leave-pot-untouched"
  src="https://embedded-journeys.moneyhelper.org.uk/en/leave-pot-untouched"
  title="leave-pot-untouched"
/>
`;
